packages/react-app/src/pages/location.tsx

```diff
@@ -0,0 +1,248 @@
+import 'leaflet/dist/leaflet.css';
+
+import { useMemo, useRef, useState } from 'react';
+import { Circle, MapContainer, Marker, TileLayer } from 'react-leaflet';
+
+import { Box, Button, Grid, Slider, Snackbar, TextField, Typography } from '@mui/material';
+import { useNavigate } from 'react-router-dom';
+import SubHeader from '../components/SubHeader';
+import { Location } from '../lib/types';
+import { Controller, useForm } from 'react-hook-form';
+import FrontendApiClient from '../services/backend';
+import MuiAlert from '@mui/material/Alert';
+
+const textInputStyle = {
+  '& .MuiInputBase-root': {
+    backgroundColor: 'white',
+    borderRadius: '8px',
+    '& fieldset': {
+      borderColor: '#E1E6EF',
+    }
+  },
+};
+const fieldErrorDescriptionStyle = {
+  color: '#FF0000',
+};
+
+const defaultPin = {
+  lat: 38.7531,
+  lng: -9.1452,
+}
+const defaultRadius = 200
+
+const LocationEdit = () => {
+  const navigate = useNavigate();
+
+  const { handleSubmit, register, formState, setValue, control } = useForm<Location>({});
+
+  const [position, setPosition] = useState(defaultPin);
+  const [mapRadius, setRadius] = useState(defaultRadius);
+
+  function DraggableMarker() {
+    const markerRef = useRef<any>(null)
+    const eventHandlers = useMemo(
+      () => ({
+        dragend() {
+          const marker = markerRef.current
+          if (marker) {
+            setPosition(marker.getLatLng())
+          }
+        },
+      }),
+      [],
+    )
+
+    return (
+      <>
+        <Circle
+          color='transparent'
+          center={position}
+          eventHandlers={eventHandlers}
+          pathOptions={{ fillColor: '#12BB6A' }}
+          radius={mapRadius}>
+        </Circle>
+        <Marker
+          draggable={true}
+          eventHandlers={eventHandlers}
+          position={position}
+          ref={markerRef}>
+        </Marker>
+      </>
+    )
+  }
+
+  const onSave = (data: Location) => {
+    FrontendApiClient.postLocation(data)
+      .then(() => {
+        navigate(-1);
+      })
+      .catch(e => {
+        setErrorMessage(e.message);
+      });
+  }
+
+  const [errorMessage, setErrorMessage] = useState<string | undefined>(undefined);
```

Comment:
Let's change to null since undefined will make us believe the state was not undefined, instead of being it defined but having an empty value
```suggestion
  const [errorMessage, setErrorMessage] = useState<string | null>(null);
```